src/sequence_state.cc

```diff
@@ -96,13 +107,35 @@ SequenceStates::Initialize(
         std::string, const inference::ModelSequenceBatching_State&>&
         state_output_config_map,
     const size_t max_batch_size,
-    const std::unordered_map<std::string, InitialStateData>& initial_state)
+    const std::unordered_map<std::string, InitialStateData>& initial_state,
+    TRITONSERVER_InstanceGroupKind kind, int32_t device_id)
 {
   input_states_.clear();
   output_states_.clear();
 
   for (auto& state : state_output_config_map) {
     auto& state_config = state.second;
+    bool use_single_buffer = state_config.use_single_buffer();
+    bool use_growable_memory = state_config.use_growable_memory();
+
+    if (use_growable_memory) {
+      if (!use_single_buffer) {
+        return Status(
+            Status::Code::INVALID_ARG,
+            std::string("In order to 'use_growable_memory', you must also "
+                        "enable 'use_single_buffer' for state input name '") +
+                state_config.input_name() + ("'."));
+      }
+      if (kind != TRITONSERVER_INSTANCEGROUPKIND_GPU) {
+        return Status(
+            Status::Code::INVALID_ARG,
+            std::string("'use_growable_memory' can only work with KIND_GPU "
+                        "model instances right now. Please either change the "
+                        "instance type to KIND_GPU or set "
+                        "'use_growable_memory' to false for state input '") +
```

Comment:
Should this be enforced to gpu instances only? What about KIND_MODEL? Or even KIND_CPU that requests cpu memory but gets gpu memory back anyways? Just curious if instance kind should enforce it here.

Reply:
Is TRT-LLM KIND_GPU or KIND_MODEL?

Reply:
For this release we will be only supporting KIND_GPU. The issue is that we need to determine the initial state memory type and device ID for each instance. I have filed a ticket to investigate this:

https://jirasw.nvidia.com/browse/DLIS-5795

Reply:
It is KIND_CPU.